README.md

```diff
@@ -1,38 +1,89 @@
 
-- If you’re building an extension for the main Checkout flow, you’ll need to read [`@shopify/argo-checkout`](packages/argo-checkout), which has all of the typings for the API available to Argo extensions in Checkout. It also has documentation for the available [global API](packages/argo-checkout/documentation/globals.md), [extension points](packages/argo-checkout/documentation/extension-points.md), and [components](packages/argo-checkout/documentation/components.md).
-- If you’re building a post-purchase extension, the package check out [`@shopify/argo-post-purchase`](packages/argo-post-purchase), which contains the same documentation noted above for the post-purchase extension points.
+### Components
+
+On top of this technical foundation, Argo provides a set of components that extensions can render. In remote-ui, the components your extension renders are actually just tiny JavaScript objects — they don’t have any DOM attached to them at all. These minimal components are sent to the host application to be rendered into native UI. This allows us to expose a minimal API for components that is focused on what you as a developer actually need to do — responding to events or customizing the appearance — while giving Shopify the ability to offer a tightly-optimized, highly-performant set of native UI components in the host application. Because your extension does not render to the actual DOM, Shopify can change or update the components without you needing to take any action, and we can make sure the components fit in seamlessly if they are in a context that allows merchant theming, like Shopify’s Checkout.
+
+The exact components available to the extension depend on the surface area you are embedding — please refer to the [documentation linked above](#i-just-want-to-build-an-argo-extension-not-learn-about-them) for references on what components you can use. If you’ve ever used [Polaris](https://polaris.shopify.com/), though, the components you will find will look extremely familiar, as many components and props are based on their Polaris equivalents.
+
+### Extension points
+
+While remote-ui provides the component model, and takes care of propagating updates to an extension’s UI into the host application, it does not have any built-in notion of an “extension” at all. To create an extension system, Shopify authors a bit of extra code: a mapping of “extension points” (which, in code, are just strings with a specific naming format), and a way to load third party code that can register to be called for those extension points. This might sound little complicated, but the code to put it all together is actually quite straightforward. The Argo ”runtime” code, which is used to construct the sandbox your extension will run in, looks almost exactly like this for all Argo-capable surface areas in Shopify:
```

Comment:
```suggestion
While remote-ui provides the component model, and takes care of propagating updates to an extension’s UI into the host application, it does not have any built-in notion of an “extension” at all. To create an extension system, Shopify authors a bit of extra code: a mapping of “extension points” (which, in code, are just strings with a specific naming format), and a way to load third-party code that can register to be called for those extension points. This might sound little complicated, but the code to put it all together is actually quite straightforward. The Argo ”runtime” code, which is used to construct the sandbox your extension will run in, looks almost exactly like this for all Argo-capable surface areas in Shopify:
```